crates/dbsp/src/operator/trace.rs

```diff
@@ -698,8 +698,8 @@ where
         self.trace = Some(i);
 
         self.dirty[0] = dirty;
-        for d in self.dirty[1..].iter_mut() {
-            *d = *d || dirty;
+        if dirty {
```

Comment:
fill doesn't do index 0?

Reply:
> fill doesn't do index 0?

It does. This could also be written as

```rust
if dirty {
    self.dirty.fill(true);
} else {
    self.dirty[0] = false;
}
```

I thought the version I submitted was marginally better.